src/shell/shell.cpp

```diff
@@ -452,6 +453,67 @@ void DOS_Shell::SyntaxError()
 	WriteOut(MSG_Get("SHELL_SYNTAX_ERROR"));
 }
 
+static std::string get_command_history_path()
+{
+	const auto section = static_cast<Section_prop *>(control->GetSection("dos"));
+	if (section) {
+		const auto path = section->Get_path("command_history_file");
+		if (path) {
+			return path->realpath;
+		}
+	}
+	return {};
+}
+
+void DOS_Shell::ReadCommandHistory()
+{
+	if (control->SecureMode()) {
+		return;
+	}
+	std::string history_path = get_command_history_path();
+	if (history_path.empty()) {
+		return;
+	}
+	std::ifstream history_file(history_path);
+	if (history_file) {
+		std::string line;
+		while (getline(history_file, line)) {
+			if (!line.empty()) {
```

Comment:
maybe we should `trim` and then check if it's empty, just to blow away useless lines too.

`void trim(std::string &str, const char trim_chars[] = " \r\t\f\n");` (from string_utils.h)

I think the prompt has a length limit (should match DOS's own length limit), but I'm not sure what that is - might have to dig in the shell code.  We should ignore lines that exceed that.

Reply:
It's a variable length thing based on how many lines and columns the user's display is.  Probably larger than anything the original DOS could do based on that description (a single command can fill up most of the screen).

https://github.com/dosbox-staging/dosbox-staging/blob/58a10bd49eeba87ed356fffd04310ff7ab163542/src/shell/shell_misc.cpp#L375-L385

Anyway, agree with checking length and trimming away useless whitespace.  I'll take a look at that tomorrow probably.

Reply:
> Probably larger than anything the original DOS could do

Interesting! Maybe we can set a reasonable line limit just for the history, (simple hard-coded limit) like the 500-item limit?

Reason is - I honestly don't want a fullscreen worth of text landing in my command history.. it would be highly suspect and likely garbage data.

I think it's easier for users to get accidental junk on their prompt (say some combination of `type`ing, `|`ing, and/or `echo`ing a binary file) versus the odds of someone manually typing a valid command sequences that sprawls for the entire screen.

(Even doing manual linux prompt entry w/ sed, awk, etc.. it's rare to need more than a couple lines).

Reply:
Sounds good.  I set a limit to 256 characters per line.